tests/demo/main.c

```diff
@@ -1,35 +1,26 @@
-#include <stdio.h>
 #include <stdlib.h>
-
-
-#define PIO_BASE 0x60000
+#include <stdint.h>
+#include "sifive_pio0.h"
```

Comment:
Two things:

1. You should include the driver headers using `<>`, not `""`. The metal driver should be considered part of the "system-installed libraries", not part of the user's application.
2. We should make sure the include path points to the block-specific driver within a directory. It looks like the example programs for the metal drivers include the drivers at [`<metal/*.h>`](https://github.com/sifive/example-watchdog/blob/be15c38fc63e3e4285c40a2d84af1b649b9ba8bf/watchdog.c#L6) , and the SiFive-specific (vendor-specific) driver implementations get placed at [`<metal/drivers/sifive_*.h>`](https://github.com/sifive/freedom-metal/blob/4c150783b6c13f715a4ee6a7ce3583f85ae5d1ce/metal/drivers/sifive_spi0.h#L9).

Can we match the conventions that the other freedom-metal software uses?

Reply:
then we would have to restructure the directory, or copy in the whole block when compiling.
what you are saying is that the -I directive on the compiler line would have to be .../block-pio-sifive.

also, metal/drivers is somehwere specific in the metal tree. Having to directories called <metal/drivers/...> is probably too confusing

I would go with making a directory pio in metal drivers, so you would do

\#include <pio/pio.h>

Reply:
I think it would be fine to restructure the directory. I didn't think too deeply about the how the include paths would work when I first suggested this directory structure, as I mostly was just thinking about getting the driver code separated from the test code.

I think what you have now works for now, but we should probably get some guidance from the freedom-metal team about how we should do this. Let's revisit after we solve the "How do we get Freedom Metal to acknowledge a new DTS node type" problem.